integr8ly/installation
# Move master-config copy from the /tmp folder to /master

## Additional Information
The original identity provider config for a cluster is currently saved in the /tmp folder during [installation](https://github.com/integr8ly/installation/blob/master/evals/roles/rhsso/tasks/identityprovider.yml#L34). This needs to be moved as the contents of the `/tmp` directory may get removed over time.

The following changes have been done for the installation:
- Save a backup copy of the original identity provider configuration from the oldest master-config backed up by ansible (`master-config.yaml.<timestamp>~`). This step is only executed if the backup copy doesn't exist or has no content
- The original identity provider config backup has now been moved to `/etc/origin/master/idp-config.integreatly-original.yaml`.

The following changes have been done for the uninstallation:
- Restoration of the identity provider config using the file `/etc/origin/master/idp-config.integreatly-original.yaml`. This will only be attempted by the uninstaller if this file exists and has content.

This PR should also fix issues in regards to subsequent attempts of running the installer and uninstaller.

## Verification Steps
1. Run the uninstall playbook 
2. Ensure this runs successfully and the identity provider content was not changed.
3. Run the installer playbook
4. Ensure that the installation completes successfully and a copy of the original identity provider can be seen in the `/etc/origin/master` directory with the correct content.
5. Run the installer playbook again.
6. Ensure that the content of the identity provider copy has not been changed.
7. Run the uninstall playbook
8. Ensure the uninstall playbook runs successfully
9. Ensure that the identity provider config was restored to it's original content.
